File: sortedArray.go
package main

import (
	"fmt"
	"sort"
)

func SortedArray(array []int) []int {
	sortedSquares := make([]int, len(array))

	for idx, value := range array {
		sortedSquares[idx] = value * value
	}
	sort.Ints(sortedSquares)
	return sortedSquares
}
func main() {
	var input1 = []int{0, 6, 7, 3}

	var x []int
	x = SortedArray(input1)
	fmt.Println(x)
}

File: sortedArray.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> sortedArray(vector<int> array)
{
    vector<int> sortedSquares(array.size(), 0);
    for (int idx = 0; idx < array.size(); idx++)
    {
        int value = array[idx];
        sortedSquares[idx] = value * value;
    }
    sort(sortedSquares.begin(), sortedSquares.end());
    return sortedSquares;
}
int main()
{

    vector<int> vect{10, 40, 30};

    vector<int> ans = sortedArray(vect);
    for (int i : ans)
    {
        cout << i << endl;
    }
    return 0;
}